tests/test_main_loop.py

```diff
@@ -0,0 +1,47 @@
+import unittest
```

Comment:
You're not calling the other test files from here, where will they be called?

Reply:
The tests can be ran from the python CLI or within an IDE, I don't have a file that calls all of them in this commit though.